test/upgrade/installation/shell.go

```diff
@@ -36,19 +37,20 @@ func LatestRelease() pkgupgrade.Operation {
 func install(installName, shellFunc string) pkgupgrade.Operation {
 	return pkgupgrade.NewOperation(installName, func(c pkgupgrade.Context) {
 		c.Log.Info("Running shell function: ", shellFunc)
-		if err := callShellFunction(shellFunc); err != nil {
+		if err := callShellFunction(shellFunc, c.T); err != nil {
 			c.T.Error(err)
 		}
 	})
 }
 
-func callShellFunction(funcName string) error {
+func callShellFunction(funcName string, t *testing.T) error {
 	loc, err := shell.NewProjectLocation("../../..")
 	if err != nil {
 		return err
 	}
 	exec := shell.NewExecutor(shell.ExecutorConfig{
 		ProjectLocation: loc,
+		Streams:         shell.TestingTStreams(t),
```

Comment:
The `ExecutorConfig` is a configuration object and I suppose it was created specifically to prevent passing too many arguments to `NewExecutor` (which it does nicely). 
As I answered in the other PR, the custom streams are really "custom" , the executor doesn't need to log into testing.T if the user wants normal stdout/stderr. So if we want to keep this shell executor generic like it was before we should not enforce logging to testing.T.